docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,197 @@
+
+### Overview
+------------
+
+File upload to S3 can also be done using **AbstractFileOutputOperator** but that operator uploads large files sequentially; the current module in contrast can substantially improve the upload speed of large files by reading and uploading their constituent blocks in parallel. 
+
+The table below lists additional benefits of this module over **AbstractFileOutputOperator**.
+
+**S3OutputModule**|**AbstractFileOutputOperator**
+-----|-----
+Maximum upload file size is 5TB.|Maximum upload file size is 5GB.
+Best fit for both large and small files.|Best fit for small files.
+Module uses AmazonS3Client API's to upload objects into S3. Large files will upload using multipart feature and small files(single block) will upload using **putObject(...)** API|Operator uses Hadoop filesystems like **S3AFileSystem**. Consists of couple of steps to upload object into S3: (1) Write the data into the local filesystem. (2) When the stream closes, filesystem uploads the local object into S3.
+If a block fails to upload then you need to re-upload the data for that block only|If a file fails to upload then you need to re-upload the complete file.
+
+### Multipart Upload Feature
+----------------------------
+Uploading the blocks of a file into S3 can be achieved by [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
+
+Using Multipart Upload feature, we can upload a single large file in multiple smaller parts. In Multipart Upload process, each part is a portion of object's data. Each part can be uploaded independently. 
+After all parts of a file are uploaded successfully, Amazon S3 combines the parts as a single object.
+
+Please refer the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### Module
+----------
+Module provides a way to use a group of operator as a single entity in a application. Module will contain list of input ports, list of output ports, configuration properties.
+Module will populate internal components properties as required based on its properties.
+
+### Operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By successfully initiating the file, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must be included in each operation like upload the blocks, multipart upload completion request of a file.
```

Comment:
initiating the file => initiating the upload
like upload the blocks, multipart upload completion request of a file. => like block upload and upload completion request.